nixos/modules/services/web-apps/pds.nix

```diff
@@ -0,0 +1,232 @@
+{
+  lib,
+  pkgs,
+  config,
+  ...
+}:
+let
+  cfg = config.services.pds;
+
+  inherit (lib)
+    getExe
+    mkEnableOption
+    mkIf
+    mkOption
+    mkPackageOption
+    escapeShellArgs
+    concatMapStringsSep
+    types
+    literalExpression
+    ;
+
+  pdsadminWrapper =
+    let
+      cfgSystemd = config.systemd.services.pds.serviceConfig;
+    in
+    pkgs.writeShellScriptBin "pdsadmin" ''
+      DUMMY_PDS_ENV_FILE="$(mktemp)"
+      trap 'rm -f "$DUMMY_PDS_ENV_FILE"' EXIT
+      env "PDS_ENV_FILE=$DUMMY_PDS_ENV_FILE"                                                   \
+          ${escapeShellArgs cfgSystemd.Environment}                                            \
+          ${concatMapStringsSep " " (envFile: "$(cat ${envFile})") cfgSystemd.EnvironmentFile} \
+          ${getExe pkgs.pdsadmin} "$@"
+    '';
+in
+# All defaults are from https://github.com/bluesky-social/pds/blob/8b9fc24cec5f30066b0d0b86d2b0ba3d66c2b532/installer.sh
+{
+  options.services.pds = {
+    enable = mkEnableOption "pds";
+
+    package = mkPackageOption pkgs "pds" { };
+
+    settings = mkOption {
+      type = types.submodule {
+        freeformType = types.attrsOf (types.nullOr types.str);
+        options = {
+          PDS_PORT = mkOption {
+            type = types.str;
```

Comment:
```suggestion
            type = types.port;
```

Reply:
That would be quite annoying actually because we have `freeformType = types.attrsOf (types.nullOr types.str);` we'd need to change it to `freeformType = types.attrsOf (types.oneOf [(types.nullOr types.str) types.port]);` and handle that later when we convert them back to strings